client/src/components/Profile/Profile.js

```diff
@@ -1,8 +1,39 @@
-import React from "react";
+import { React, useState } from "react";
 import PropTypes from "prop-types";
+import * as fetch from "../../lib/fetch";
 
 function Profile({ id }) {
-  return <>Profile of user with id {id}</>;
+  const [userExists, setUserExists] = useState(true);
+  const [name, setName] = useState("");
+  const [userID, setUserID] = useState("");
+
+  const getData = async () => {
+    const { ok, resJSON } = await fetch.getUserData(id);
+    return { ok, resJSON };
+  };
+
+  getData().then(({ ok, resJSON }) => {
+    if (!ok) {
+      setUserExists(false);
+      return;
+    }
+    setName(resJSON.name);
+    setUserID(resJSON.id);
+  });
+
+  const renderUser = () => {
+    return (
+      <>
+        Name is {name}, ID is {userID}
+      </>
+    );
+  };
+
+  const renderNotFound = () => {
+    return <>Specified user was not found.</>;
+  };
+
+  return <>{userExists ? renderUser() : renderNotFound()}</>;
```

Comment:
Thinking out loud, do we want to have a "loading" state?

Reply:
Yes, we will definitely want some sort of loading state in probably a lot of pages. I was thinking we could create some sort of `<Loading>` component to render during such a state, but was too lazy to do so in this PR :')